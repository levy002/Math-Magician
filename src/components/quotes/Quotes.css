.quotes-Container {
  background-color: rgba(195, 251, 255, 0.8);
  padding: 50px;
  border-radius: 20px;
  display: flex;
  flex-flow: column;
  gap: 25px;
  margin: 0 10vh;
}

.quotes-Container li {
  display: flex;
  flex-flow: column;
  gap: 3px;
}

.quotes-Container li p {
  margin: 0;
}

.quote {
  font-size: 1.2rem;
  font-weight: bold;
  font-style: italic;
}

.creator {
  color: rgb(66, 6, 163);
  font-weight: bold;
  font-style: italic;
}

@media only screen and (max-width: 768px) {
  .quotes-Container {
    margin: 0 3vh;
    padding: 30px;
  }

  .quotes-Container :nth-child(5) {
    display: none;
  }
}
