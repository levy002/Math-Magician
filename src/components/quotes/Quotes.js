import React from 'react';
import { v4 as uuidv4 } from 'uuid';
import './Quotes.css';

const Quotes = () => {
  const randomQuotes = [
    {
      quote: 'The essence of mathematics resides in its freedom.',
      id: uuidv4,
      creator: 'Georg Cantor',
    },
    {
      quote: 'Without mathematics, thereâ€™s nothing you can do. Everything around you is mathematics. Everything around you is numbers.',
      id: uuidv4,
      creator: 'Shakuntala Devi',
    },
    {
      quote: 'Mathematics is the most beautiful and most powerful creation of the human spirit.',
      id: uuidv4,
      creator: 'Stefan Banach',
    },
    {
      quote: 'Mathematics as an expression of the human mind reflects the active will, the contemplative reason, and the desire for aesthetic perfection. Its basic elements are logic and intuition, analysis and construction, generality and individuality.',
      id: uuidv4,
      creator: 'Richard Courant',
    },
    {
      quote: 'As far as the laws of mathematics refer to reality, they are not certain, and as far as they are certain, they do not refer to reality.',
      id: uuidv4,
      creator: 'Albert Einstein',
    },
  ];

  return (
    <ul className="quotes-Container">
      {randomQuotes.map((q) => (
        <li key={q.id}>
          <p className="quote">
            &#34;
            {q.quote}
            &#34;
          </p>
          <p className="creator">{q.creator}</p>
        </li>
      ))}
    </ul>
  );
};

export default Quotes;
