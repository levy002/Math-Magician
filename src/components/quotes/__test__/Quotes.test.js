import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Quotes from '../Quotes';

test('renders "Albert Einstein" on Quotes link', () => {
  render(<Quotes />);
  const paragraphElement = screen.getByText(/georg/i);
  expect(paragraphElement).toBeTruthy();
});

it('renders correctly', () => {
  const tree = renderer.create(<Quotes />).toJSON();
  expect(tree).toMatchSnapshot();
});
