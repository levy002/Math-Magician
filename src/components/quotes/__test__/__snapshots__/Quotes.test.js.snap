// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="quotes-Container"
>
  <li>
    <p
      className="quote"
    >
      "
      The essence of mathematics resides in its freedom.
      "
    </p>
    <p
      className="creator"
    >
      Georg Cantor
    </p>
  </li>
  <li>
    <p
      className="quote"
    >
      "
      Without mathematics, thereâ€™s nothing you can do. Everything around you is mathematics. Everything around you is numbers.
      "
    </p>
    <p
      className="creator"
    >
      Shakuntala Devi
    </p>
  </li>
  <li>
    <p
      className="quote"
    >
      "
      Mathematics is the most beautiful and most powerful creation of the human spirit.
      "
    </p>
    <p
      className="creator"
    >
      Stefan Banach
    </p>
  </li>
  <li>
    <p
      className="quote"
    >
      "
      Mathematics as an expression of the human mind reflects the active will, the contemplative reason, and the desire for aesthetic perfection. Its basic elements are logic and intuition, analysis and construction, generality and individuality.
      "
    </p>
    <p
      className="creator"
    >
      Richard Courant
    </p>
  </li>
  <li>
    <p
      className="quote"
    >
      "
      As far as the laws of mathematics refer to reality, they are not certain, and as far as they are certain, they do not refer to reality.
      "
    </p>
    <p
      className="creator"
    >
      Albert Einstein
    </p>
  </li>
</ul>
`;
